blockmanager.go

```diff
@@ -1393,68 +1393,64 @@ func resolveCFHeaderMismatch(block *wire.MsgBlock, fType wire.FilterType,
 	// network, we're forced to only verify what we can at this point. So
 	// we'll just ensure that each of the filters returned
 	//
 	// TODO(roasbeef): update after BLOCK_WITH_PREV_OUTS is a thing
-	case wire.GCSFilterRegular:
-
-		// We'll now run through each peer and ensure that each output
-		// script is included in the filter that they responded with to
-		// our query.
-		for peerAddr, filter := range filtersFromPeers {
-		peerVerification:
-
-			// We'll ensure that all the filters include every
-			// output script within the block.
-			//
-			// TODO(roasbeef): eventually just do a comparison
-			// against decompressed filters
-			for _, tx := range block.Transactions {
-				for _, txOut := range tx.TxOut {
-					switch {
-					// If the script itself is blank, then
-					// we'll skip this as it doesn't
-					// contain any useful information.
-					case len(txOut.PkScript) == 0:
-						continue
-
-					// We'll also skip any OP_RETURN
-					// scripts as well since we don't index
-					// these in order to avoid a circular
-					// dependency.
-					case txOut.PkScript[0] == txscript.OP_RETURN:
-						continue
-					}
 
-					match, err := filter.Match(
-						filterKey, txOut.PkScript,
-					)
-					if err != nil {
-						// If we're unable to query
-						// this filter, then we'll skip
-						// this peer all together.
-						continue peerVerification
-					}
-
-					if match {
-						continue
-					}
+	// We'll now run through each peer and ensure that each output
+	// script is included in the filter that they responded with to
+	// our query.
+	for peerAddr, filter := range filtersFromPeers {
+		// We'll ensure that all the filters include every output
+		// script within the block.
+		//
+		// TODO(roasbeef): eventually just do a comparison against
+		// decompressed filters
+	peerVerification:
+		for _, tx := range block.Transactions {
+			for _, txOut := range tx.TxOut {
+				switch {
+				// If the script itself is blank, then we'll
+				// skip this as it doesn't contain any useful
+				// information.
+				case len(txOut.PkScript) == 0:
+					continue
+
+				// We'll also skip any OP_RETURN scripts as
+				// well since we don't index these in order to
+				// avoid a circular dependency.
+				case txOut.PkScript[0] == txscript.OP_RETURN:
+					continue
+				}
 
-					// If this filter doesn't match, then
-					// we'll mark this peer as bad and move
-					// on to the next peer.
-					badPeers[peerAddr] = struct{}{}
+				match, err := filter.Match(
+					filterKey, txOut.PkScript,
+				)
+				if err != nil {
+					// If we're unable to query this
+					// filter, then we'll skip this peer
+					// all together.
 					continue peerVerification
 				}
+
+				if match {
+					continue
+				}
+
+				// If this filter doesn't match, then we'll
+				// mark this peer as bad and move on to the
+				// next peer.
+				badPeers[peerAddr] = struct{}{}
+				continue peerVerification
```

Comment:
Seems like this should be `break` instead, otherwise we'll continue to check the block again?

Reply:
Good catch! I agree, fixed by moving label to outer loop: https://github.com/lightninglabs/neutrino/pull/140/commits/f5b07e94e6d7be48cd1c747fca1fae1e398ab9cf